app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomViewBaseState.kt

```diff
@@ -24,32 +24,40 @@ class MaskCustomViewBaseState(
 
     internal val basePathsAndPaintChangesData = extractBaseData(pathsPaintsAndStagesNames)
 
-    private fun extractBaseData(inputList: List<MaskLayer.PathPaintAndStageName>): List<BaseData> {
+    private fun extractBaseData(inputList: List<MaskLayer.PathAndPaint>): List<BaseData> {
         val result = LinkedList<BaseData>()
         if (inputList.isNotEmpty()) {
-            val (firstPath, firstPaint, firstStageName) = inputList[0]
+            val (firstPath, firstPaint) = inputList[0]
             val firstPathBaseData = PathBaseData(firstPath.points)
             val firstPaintChangeBaseData =
-                PaintChangeBaseData(firstPaint.color, firstPaint.strokeWidth)
-            result.add(BaseData(firstPathBaseData, firstPaintChangeBaseData, firstStageName))
+                PaintChangeBaseData(
+                    firstPaint.color,
+                    firstPaint.strokeWidth,
+                    firstPaint.xfermode == MaskLayer.ERASER_XFER_MODE
+                )
+            result.add(BaseData(firstPathBaseData, firstPaintChangeBaseData))
             var latestPaint: Paint = firstPaint
-            for ((path, paint, stageName) in inputList.subList(1, inputList.size)) {
+            for ((path, paint) in inputList.subList(1, inputList.size)) {
                 val pathPoints = path.points
                 val pathBaseData = PathBaseData(pathPoints)
                 val paintBaseChangeData =
-                    if (paint != latestPaint)
-                        PaintChangeBaseData(paint.color, paint.strokeWidth)
-                    else
+                    if (paint == latestPaint)
                         null
```

Comment:
It's because it's actually a ChangeBaseData, so if there was no change since the previous entry, nothing is put there. Just the way I designed it.

Reply:
So for all the `PathBaseData` (which are basically the same as a `PointToPointPath`) we have a 'change' object which has a color, width, and whether it is an eraser. Which we then use to do... what exactly? 

I'm sorry man but this bit is a little confusing to me

Reply:
Don't worry at all!! So, the point of these objects is to tackle a problem we had with screen rotations: When the screen is rotated, Android recreates all objects - before it, though, it saves the objects states; after recreating them, it asks the newly created objects to restore their states with these "state objects" it saved. These objects you're seeing are the "state objects". We didn't use the Paths directly because it's not clear whether their insides have things that would use more memory than needed - instead we extract from the only the path points, and with those we recreate the Path object on state restoration. The same goes to the Paints: instead of storing the whole objects, we store what actually changes in them (if anything) and recreate them when needed. Does it make more sense now?

Reply:
OK cool I get what's going on now. Still feels a bit shaky to me but since the behaviour doesn't fundamentally change in this PR it should be good 👍